docs/getting-started-guides/kubeadm.md

```diff
@@ -177,12 +177,43 @@ You must install a pod network add-on so that your pods can communicate with eac
 
 **It is necessary to do this before you try to deploy any applications to your cluster, and before `kube-dns` will start up. Note also that `kubeadm` only supports CNI based networks and therefore kubenet based networks will not work.**
 
-Several projects provide Kubernetes pod networks using CNI, some of which 
-also support [Network Policy](/docs/user-guide/networkpolicies/). See the [add-ons page](/docs/admin/addons/) for a complete list of available network add-ons.
-
-You can install a pod network add-on with the following command: 
-
-    # kubectl apply -f <add-on.yaml>
+Several projects provide Kubernetes pod networks using CNI, some of which also support [Network Policy](/docs/user-guide/networkpolicies/).
+
+You must select a network addon from the drop-down menu below, and correct instructions will appear below.
+
+See the [add-ons page](/docs/admin/addons/) to learn more about each of the available network add-ons.
+
+<script>
+  var networkAddons = {
+    "Calico": "kubectl apply -f \"http://docs.projectcalico.org/v2.0/getting-started/kubernetes/installation/hosted/calico.yaml\"",
+    "Flannel": "kubeclt apply -f \"https://github.com/coreos/flannel/blob/master/Documentation/kube-flannel.yml?raw=true\"",
+    "Weave Net": "kubectl apply -f \"https://git.io/weave-kube\"",
+  };
+
+  $(document).ready(function(){
+    var code = $('#networkAddonInstallCommand')
+      .children('pre')
+      .children('code');
+    var menu = $('#pickNetworkAddon').first();
+    for( var val in networkAddons ) {
+      $('<option />', { text: val, value: networkAddons[val] }).appendTo(menu);
+    }
+    menu.change(function(ev) {
+      code.text("# " + ev.target.value);
+    });
+  });
+</script>
+
+<p>
+  <select id="pickNetworkAddon" style="margin:1em;font-size:1.2em;">
+    <option>Please select a network addon...</option>
```

Comment:
In case the user chooses this, I think the command should go back to `# kubectl apply -f <selected-add-on.yaml>`, now it does not